### PR TITLE
feat: link node

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

### Issue

Currently it's a bit hard to show to our users what links their buttons goes to

[Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/37071143/todos/7292062642)

### Solution

We are to create a link node to better inform our users about their URLs

- Create link nodes which attach and follow `ActionButton`'s
- Removed icons from left of the ActionButton (it was the temporary solution)
- `LinkNode` positions are calculated based of `StepBlockNode` position as well as number of actions on the step
- Refactor / tidy up for lib files with logic calculating goals
- Refactor `BaseNode` so handles can be disabled (handle renders, but not connectable)

### CODE REVIEW

**Review by scottkelso — COMMENTED** [review]
Needs some unit tests but otherwise looking good! Just some thoughts - nothing major.

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/ActionButton/ActionButton.tsx [hunk 1]
```diff
@@ -27,78 +28,45 @@ export function ActionButton({
...
+  function getTitle(block, defaultTitle): string {
+    if (block.label != null && block.label !== '') return block.label
+    if (block.__typename === 'VideoBlock')
+      return block.video?.title?.[0]?.value ?? block.title ?? t('Video')
+    return defaultTitle
   }
 
-  function hasConnection(block): boolean {
-    return (
+  function extractTitleAndConnection(block, defaultTitle): BlockUIProperties {
```
[the remark this answers is not in the document]
Comment (GeronimoJohn) [reply apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/ActionButton/ActionButton.tsx:38]: that makes sense, will work on that later on!

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/libs/transformSteps/transformSteps.ts [hunk 2]
```diff
@@ -99,12 +104,41 @@ export function transformSteps(
         ...defaultEdgeProps
       })
     }
+
+    if (
+      block.action.__typename === 'LinkAction' ||
+      block.action.__typename === 'EmailAction'
+    ) {
+      edges.push({
+        id: `${block.id}->LinkNode-${block.id}`,
```
[the remark this answers is not in the document]
Comment (GeronimoJohn) [reply apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/libs/transformSteps/transformSteps.ts:113]: The first one! Its the syntax from ReactFlow, indicates connection

File: libs/journeys/ui/src/components/Button/utils/getLinkActionGoal/getLinkActionGoal.ts [hunk 3]
```diff
@@ -22,15 +22,25 @@ export enum GoalType {
   Chat = 'Chat',
```
[the remark this answers is not in the document]
Comment (GeronimoJohn) [reply libs/journeys/ui/src/components/Button/utils/getLinkActionGoal/getLinkActionGoal.ts:22]: Yea that makes sense

**Review by levi-zustiak — APPROVED** [review]

**Pull request merged**